src/sdk/namespace_fs.js

```diff
@@ -2386,7 +2388,10 @@ class NamespaceFS {
         try {
             await nb_native().fs.stat(fs_context, this.bucket_path);
         } catch (err) {
-            throw this._translate_object_error_codes(err);
+            const err_to_throw = this._translate_object_error_codes(err);
+            // this a temp solution so on the function related to bucket we will change the error to the relevant one (bucket instead of object)
+            if (err_to_throw && err_to_throw.rpc_code === 'NO_SUCH_OBJECT') err_to_throw.rpc_code = 'NO_SUCH_BUCKET';
```

Comment:
@shirady Will it fail for object? for object error code will be `NO_SUCH_BUCKET`?

Reply:
As I understand it we are in a function that it's name is `_load_bucket`, when it received the `ENOENT` error code it used `_translate_object_error_codes` and it was `NO_SUCH_OBJECT` and I don't think it supposed to be the rpc_code, because the action it on a bucket and not object.

> @shirady Will it fail for object? for object error code will be `NO_SUCH_BUCKET`?

I'm not sure I understand the question, I can take an example maybe - if someone tries to delete an object and the path of the bucket was deleted the flow should be: delete_object -> _load_bucket (failed, cannot stat the path) -> throw err with rpc_code 'NO_SUCH_BUCKET' -> won't be translated again in `_translate_object_error_codes`. I hope this answers your question.

Reply:
@shirady my doubt is,  When try to list the object which dosn't exists inside bucket, will this code return `NO_SUCH_BUCKET`?

Reply:
I'm not sure I understand.
I tested listing a non-existing object in AWS and it doesn't throw an error (and we also don't throw an error).
I used this command: `aws s3 ls s3://<bucket-name>/<non-existing-object>`.